File: backend/api/schemas/hitl.py
from typing import Literal

from pydantic import BaseModel, Field


class HITLSubmitRequest(BaseModel):
    conversation_id: str

    reason: str = Field(
        min_length=5,
        max_length=2000,
    )


class HITLReviewRequest(BaseModel):

    status: Literal[
        "approved",
        "rejected",
    ]

    review_comment: str | None = Field(
        default=None,
        max_length=2000,
    )

File: frontend/src/types/hitl.ts
export type HITLStatus =
  | 'pending'
  | 'approved'
  | 'rejected';

export type HITLExecutionStatus =
  | 'executing'
  | 'successful'
  | 'failed'
  | null;

export interface HITLRequest {
  id: string;

  conversation_id: string;

  employee_id: string;

  reason: string;

  status: HITLStatus;

  reviewed_by: string | null;

  review_comment: string | null;

  created_at: string;

  reviewed_at: string | null;

  execution_status: HITLExecutionStatus;

  execution_result:
    | Record<string, unknown>
    | null;

  executed_at: string | null;
}

export interface HITLSubmitRequest {
  conversation_id: string;
  reason: string;
}

export interface HITLReviewRequest {
  status: 'approved' | 'rejected';
  review_comment: string | null;
}

export interface HITLResponse {
  status: string;
  hitl_request: HITLRequest;
}

export interface HITLListResponse {
  status: string;
  requests: HITLRequest[];
}